mlflow/metrics/genai/model_utils.py

```diff
@@ -99,66 +98,44 @@ def _call_openai_api(openai_uri, payload, eval_parameters, extra_headers, proxy_
         )
 
     from mlflow.openai import _get_api_config
-    from mlflow.openai.api_request_parallel_processor import process_api_requests
     from mlflow.utils.openai_utils import _OAITokenHolder
 
     api_config = _get_api_config()
     api_token = _OAITokenHolder(api_config.api_type)
+    api_token.refresh()
 
-    payload = {
-        "messages": [{"role": "user", "content": payload}],
-        **eval_parameters,
-    }
+    if api_config.api_type == "azure":
+        from openai import AzureOpenAI
 
-    if api_config.api_type in ("azure", "azure_ad", "azuread"):
-        api_base = getattr(api_config, "api_base")
-        api_version = getattr(api_config, "api_version")
-        engine = getattr(api_config, "engine")
-        deployment_id = getattr(api_config, "deployment_id")
-
-        if engine:
-            # Avoid using both parameters as they serve the same purpose
-            # Invalid inputs:
-            #   - Wrong engine + correct/wrong deployment_id
-            #   - No engine + wrong deployment_id
-            # Valid inputs:
-            #   - Correct engine + correct/wrong deployment_id
-            #   - No engine + correct deployment_id
-            if deployment_id is not None:
-                _logger.warning(
-                    "Both engine and deployment_id are set. Using engine as it takes precedence."
-                )
-            payload = {"engine": engine, **payload}
-        elif deployment_id is None:
-            raise MlflowException(
-                "Either engine or deployment_id must be set for Azure OpenAI API",
-            )
-        payload = payload
-
-        request_url = proxy_url or (
-            f"{api_base}/openai/deployments/{deployment_id}"
-            f"/chat/completions?api-version={api_version}"
-        )
+        client = AzureOpenAI(
+            api_key=api_token.token,
+            azure_endpoint=api_config.api_base,
+            api_version=api_config.api_version,
+            azure_deployment=api_config.deployment_id,
+        ).with_options(max_retries=api_config.max_retries, timeout=api_config.timeout)
     else:
-        payload = {"model": openai_uri, **payload}
-        request_url = proxy_url or REQUEST_URL_CHAT
+        from openai import OpenAI
+
+        client = OpenAI(
+            api_key=api_token.token,
+            base_url=api_config.api_base,
+        ).with_options(max_retries=api_config.max_retries, timeout=api_config.timeout)
+
+    if proxy_url is not None:
+        client.base_url = proxy_url
```

Comment:
Can you test if setting `proxy_url` works for any URL, say Databricks model serving endpoint? https://docs.databricks.com/en/machine-learning/model-serving/score-foundation-models.html#query-a-chat-completion-model


Sth like this:
```
os.environ["OPENAI_API_KEY"] = <your-databricks-access-token>

eval_data = pd.DataFrame(
    {
        "inputs": ["What is MLflow?"],
        "outputs": ["MLflow is an open-source platform for managing machine learning processes. It provides tools for tracking experiments, packaging code into reproducible runs, and sharing and deploying models."],
        "ground_truth": ["MLflow is an OSS platform for managing ML lifecycle."],
    }
)

with mlflow.start_run():
    mlflow.evaluate(
        data=eval_data,
        extra_metrics=[
            answer_similarity(
                model="openai:/
databricks-meta-llama-3-1-70b-instruct",
                proxy_url="https://example.staging.cloud.databricks.com/serving-endpoints"
            )
        ]
```

Reply:
This does not work well if the proxy API does not follow the path design of OpenAI. The method of SDK construct the endpoint by concatenating base url with "/chat/completion".
https://platform.openai.com/docs/models#model-endpoint-compatibility

Reply:
Discussed offline: This code will be migrated again to gateway+REST soon ([ref](https://github.com/mlflow/mlflow/blob/master/mlflow/metrics/genai/model_utils.py#L58)) so we decided not to support proxy URL in non OpenAI format for now.